src/javascript/app/pages/signup.js

```diff
@@ -0,0 +1,85 @@
+const BinarySocket   = require('../base/socket');
+const FormManager    = require('../common/form_manager');
+const Login          = require('../../_common/base/login');
+const getElementById = require('../../_common/common_functions').getElementById;
+const localize       = require('../../_common/localize').localize;
+
+const Signup = (() => {
+    let clients_country,
+        $google_btn,
+        $login_btn,
+        $error_msg,
+        $verify_email;
+
+    const form_id = '#signup_form';
+
+    const onLoad = () => {
+        getElementById('footer').setVisibility(0); // always hide footer in this page
+
+        BinarySocket.wait('website_status').then((response) => {
+            clients_country = response.website_status.clients_country;
+
+            FormManager.init(form_id, [
+                { selector: '#email', validations: ['req', 'email'], request_field: 'verify_email' },
+                { request_field: 'type', value: 'account_opening' },
+            ]);
+            FormManager.handleSubmit({
+                form_selector       : form_id,
+                fnc_response_handler: verifyEmailHandler,
+                fnc_additional_check: checkCountry,
+            });
+            $google_btn   = $('#google-signup');
+            $login_btn    = $('#login');
+            $verify_email = $('#verify_email');
+            $error_msg    = $('.error-msg');
```

Comment:
we should be able to declare all of these and call form manager without having to wait for website_status, so on line 18 or so, everything except for line 20 should be able to move out